week7-dom/index.html

```diff
@@ -0,0 +1,193 @@
+  y = y + Math.round(random(-1, 1));
+  
+  // update element p's content
+  p.________('x: ' + x + '<br />y: ' + y );
+  
+  // change element p's background color
+  p.________('________', '________');
+  
+  // add paddings to element p
+  p.________('________', '________');
+}
+</script>
+
+<p>
+  <iframe src="https://alpha.editor.p5js.org/embed/rJLDPApA" width=200 height=270></iframe>
+  <br/>
+
+  2. Write a program that implements the above sketch. Create a <code>button</code> element. Attach a callback function to the button's <code>mousePressed</code> event. In the callback function, give the canvas a random background color. Similarlly, create an <code>input</code> element to update some text on the canvas, and create a <code>slider</code> element to control the size of an ellipse.</p>
+
+<script type="text/p5" data-preview-width="400">
+// Please copy and paste the code to the p5 web editor to do the following questions.
+var bgcolor;
+var button;
+var slider;
+var nameInput;
+var nameP;
+
+function setup() {
+  canvas = createCanvas(200, 200);
+  bgcolor = color(200);
+  
+  // Create a button element
+  button = ________("___________");
+  // Attach a callback function called changeColor to the button's mousePressed event
+  button.________(________);
+  
+  // Create a slider element
+  slider = ________(____, ____, ____);
+
+  // Create an input element
+  nameInput = ________('________');
+  // Attach a callback function called updateText to the input's changed event
+  nameInput.________(________);
+  
+  // Create a p element
+  nameP = ________('Your name!'); 
+  // Attach a callback function called overpara to the p's mouseOver event
+  nameP.________(________);
+  // Attach a callback function called outpara to the p's mouseOut event
+  nameP.________(________);
+}
+
+// Callback function for the button's mousePressed event
+function changeColor() {
+  bgcolor = color(random(150, 255), random(150, 255), random(150, 255));
+}
+
+// Callback function for the input's changed event
+function updateText() {
+  nameP.html(nameInput.value());
+}
+
+// Callback function for the nameP's mouseOver event
+function overpara() {
+  nameP.html('your moues is over me');
+}
+
+// Callback function for the nameP's mouseOut event
+function outpara() {
+  nameP.html('your mouse is out');
+}
+
+function draw() {
+  background(bgcolor);
+
+  text(nameInput.value(), 10, 20);
+
+  noStroke(); 
+  fill(255, 50, 150);
+  // Draw an ellipse according to slider's value
+  ellipse(100, 100, ________, ________);
+}
+</script>
+
+
+<p>
+  <iframe src="https://alpha.editor.p5js.org/embed/B1fWdyCA" width=200 height=200></iframe>
+  <br/>
+
+  3. Write a program that implements the above sketch. Create multiple <code>p</code> elements. Manipulate their x, y positions. Attach callback functions to their <code>mouseOver</code> and <code>mouseOut</code> events. In the callback functions, remember to use "this" to reference the current element.</p>
```

Comment:
Here we could create a simple question about `select()` and an `id`.  Just selecting a single paragraph element to show the difference between `select(#id)` vs `createP()`.